types/recommendation.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CStorPoolRecommendation is a kubernetes custom
+// resource that represents recommended configurations
+// for one given input/requirement.
+type CStorPoolRecommendation struct {
```

Comment:
We shall try avoiding Pool in the name. Try fitting cluster instead of pool.
i.e. CStorClusterRecommendation